ludwig/api.py

```diff
@@ -1210,8 +1202,8 @@ def collect_activations(
 
         # preprocessing
         logger.debug("Preprocessing")
-        dataset, training_set_metadata = preprocess_for_prediction(
-            self.config,
+        dataset, training_set_metadata = preprocess_for_prediction(  # TODO (Connor): Refactor to use self.config_obj
+            self.config_obj.to_dict(),
```

Comment:
This is a good idea and I see the value in this, however, I am hesitant to make this change for a couple of reasons: The first and most clear reason in my opinion is because there are small edits made to the Config Object as it passes through the pipeline (preprocessing, auto_tune LR and BS, etc.) Since these changes are being made on the Config Object and not the stateful class attribute, we would need to make the same changes in parallel to the config_dict which would probably be a little more effort I think. The second reason is that the end goal of the config object is to completely replace the need for a config dict altogether (except for the user entry point), so these `config_obj.to_dict()` calls are essentially a paper trail of refactors we still need to make and will eventually just get swapped out for `config_obj` once that section of the pipeline has been refactored. For these two reasons, I feel like it may be better to leave the `to_dict()` calls around as they will slowly disappear after my followup PRs. WDYT?

Reply:
Works for me, assuming `to_dict()` is not a particularly expensive cost, which it doesn't seem like it should be.